### PR TITLE
Made sure not to turn on wide gamut support without impeller.

### PR DESCRIPTION
**Pull request by gaaclarke**
fixes https://github.com/flutter/flutter/issues/125430

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterEngine.mm [hunk 1]
```diff
@@ -1402,6 +1402,10 @@ - (FlutterDartProject*)project {
   return _dartProject.get();
 }
 
+- (BOOL)isUsingImpeller {
+  return self.project.isImpellerEnabled;
```
Comment (dnfield) [inline shell/platform/darwin/ios/framework/Source/FlutterEngine.mm:1406]: This is available via the shell, but there's nothing inherently wrong with exposing it on the project either.

File: shell/platform/darwin/ios/framework/Source/FlutterView.mm [hunk 2]
```diff
@@ -49,6 +36,23 @@ - (instancetype)initWithCoder:(NSCoder*)aDecoder {
...
+#endif
+
+  if (![_delegate isUsingImpeller]) {
+    return NO;
+  }
+
+  // This predicates the decision on the capabilities of the iOS device's
+  // display.  This means external displays will not support wide gamut if the
+  // device's display doesn't support it.  It practice that should be never.
+  return UIScreen.mainScreen.traitCollection.displayGamut != UIDisplayGamutSRGB;
```
Comment (dnfield) [inline shell/platform/darwin/ios/framework/Source/FlutterView.mm:53]: nit: This should be using availability checks and then checking the `UIScene` API on newer iOS and falling back to this deprecated property. That would also solve the problem of not knowing whether a connected display is wide gamut or not.

**Review by dnfield — APPROVED** [review]
LGTM with nits. We should file a bug to follow up on using UIScene here

File: shell/platform/darwin/ios/framework/Source/FlutterView.mm (hunk 2 above)
Comment (gaaclarke) [reply shell/platform/darwin/ios/framework/Source/FlutterView.mm:53]: Filed issue: https://github.com/flutter/flutter/issues/125496

Comment (jmagman) [reply shell/platform/darwin/ios/framework/Source/FlutterView.mm:53]: Can you instead use `self.window.windowScene.screen.traitCollection.displayGamut`? See `mainScreenIfViewLoaded` in the FlutterViewController for a fallback example.

Comment (gaaclarke) [reply shell/platform/darwin/ios/framework/Source/FlutterView.mm]: Done

**Pull request merged**